Automattic/jetpack
# Jetpack Boost: Strict TypeScript

Guardrails off.

IIRC we enabled typescript without strict mode a long time ago because we were "ramping up slowly". I think 2 years is slow enough 😛 

On top of that, this addresses missing React component types.

## Proposed changes:
Enables strict mode in TypeScript.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:

Apply patch, create a function with unspecified types, see squiggly lines:

![2023_12_13RaalHoSH 42](https://github.com/Automattic/jetpack/assets/988095/61f88b43-d718-4451-b9e3-5e7e6080f926)
